internal/common/blockstring.go

```diff
@@ -0,0 +1,77 @@
+package common
```

Comment:
This code is adapted from the graphql-js implementation, which is under the MIT license. This will need attribution as per that license, which I assume is preferred via a comment at the top of the file here?

Reply:
@dackroyd, please add a comment at the top of the file that this was translated from the reference implementation and add a link to the original code. I think that this should be enough

Reply:
I've added in the MIT License as a header to the file, (copied from https://github.com/graphql/graphql-js/blob/5eb7c4ded7ceb83ac742149cbe0dae07a8af9a30/LICENSE), along with a comment explaining that the implementation here is derived from that reference implementation.